packages/transaction-controller/src/types.ts

```diff
@@ -1307,6 +1307,9 @@ export type SimulationError = {
 
 /** Simulation data for a transaction. */
 export type SimulationData = {
+  /** Set to true if transaction re-simulated due to security concerns */
+  isReSimulatedDueToSecurity?: boolean;
```

Comment:
Are we only setting this when the response changes on a subsequent re-simulation?

Perhaps `isUpdatedAfterSecurityCheck` is more explicit?

Reply:
Done.